action/protocol/staking/protocol.go

```diff
@@ -690,17 +691,23 @@ const (
 func (p *Protocol) settleAction(
 	ctx context.Context,
 	sm protocol.StateManager,
+	act action.TxDynamicGas,
 	status uint64,
 	logs []*action.Log,
 	tLogs []*action.TransactionLog,
 	gasConsumed uint64,
 	gasToBeDeducted uint64,
 	updateNonce nonceUpdateType,
 ) (*action.Receipt, error) {
-	actionCtx := protocol.MustGetActionCtx(ctx)
-	blkCtx := protocol.MustGetBlockCtx(ctx)
-	gasFee := big.NewInt(0).Mul(actionCtx.GasPrice, big.NewInt(0).SetUint64(gasToBeDeducted))
-	depositLog, err := p.helperCtx.DepositGas(ctx, sm, gasFee)
+	var (
+		actionCtx = protocol.MustGetActionCtx(ctx)
+		blkCtx    = protocol.MustGetBlockCtx(ctx)
+	)
+	gasFee, baseFee, err := protocol.SplitGas(ctx, act, gasToBeDeducted)
```

Comment:
take care of multiple protocol handling(e.g. MigrateStake), `baseFee` maybe double-deducted

Reply:
should be OK, the `gasToBeDeducted ` here is only for the second part of action
the gas of first part (create NFT) is handled in EVM protocol